plugins/outputs/iotdb/iotdb_test.go

```diff
@@ -0,0 +1,305 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package iotdb
+
+import (
+	"fmt"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/apache/iotdb-client-go/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const testPort = "6667"
+
+func newTestClient() *IoTDB {
+	testClient := newIoTDB()
+	testClient.Host = "localhost"
+	testClient.Port = testPort
+	testClient.User = "root"
+	testClient.Password = "root"
+	testClient.Log = testutil.Logger{}
+	return testClient
+}
+
+// This func makes sure fields in order. "metric.New" uses map[string]interface{} so fields are not in order.
+func generateTestMetric(
+	name string,
+	tags []telegraf.Tag,
+	fields []telegraf.Field,
+	timestamp time.Time,
+) telegraf.Metric {
+	m := metric.New(name, map[string]string{}, map[string]interface{}{}, timestamp)
+	for _, tag := range tags {
+		m.AddTag(tag.Key, tag.Value)
+	}
+	for _, field := range fields {
+		m.AddField(field.Key, field.Value)
+	}
+	return m
+}
```

Comment:
You don't need this wrapper. Please use [`metric.New()`](https://github.com/influxdata/telegraf/blob/8003e66f1eb12358d591ad0348f2887ed01dedae/metric/metric.go#L21) instead which is equivalent.

Reply:
I think that this wrapper is really important for testing. @srebhan 
For last discussion please see [here](https://github.com/influxdata/telegraf/pull/11557#discussion_r933308843).
I need this wrapper because `metric.New()` can NOT make sure fields in order. But it's really important to testing because I have to specify the expected `RecordsWithTags`. Different orders lead to `require.EqualValues` raising an error.

Telegraf does sort tags, but does not sort fields. ([Source code](https://github.com/influxdata/telegraf/blob/8003e66f1eb12358d591ad0348f2887ed01dedae/metric/metric.go#L43))
But function [`AddField()`](https://github.com/influxdata/telegraf/blob/8003e66f1eb12358d591ad0348f2887ed01dedae/metric/metric.go#L189) makes sure fields in order. That's why I used it.

The SQL output plugin also uses a wrapper [`stableMetric()`](https://github.com/influxdata/telegraf/blob/8003e66f1eb12358d591ad0348f2887ed01dedae/plugins/outputs/sql/sql_test.go#L50) like this for testing.

Reply:
Ok, then please add this comment to the code. Additionally I would then rename that function to `newMetricWithOrderedFields()`. What do you think?

Reply:
All right! I think `newMetricWithOrderedFields()` is more reasonable and easy to read. Thanks.